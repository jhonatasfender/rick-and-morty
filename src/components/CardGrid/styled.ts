import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 1rem;
  padding: 1rem;
`
