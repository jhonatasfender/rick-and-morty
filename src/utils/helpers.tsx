import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { render, RenderResult } from '@testing-library/react'

import theme from '../styles/theme'

const queryClient = new QueryClient()

export const renderWithThemeAndRoute = (children: ReactNode): RenderResult =>
  render(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </MemoryRouter>
    </QueryClientProvider>,
  )
