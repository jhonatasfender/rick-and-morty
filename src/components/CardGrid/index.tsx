import { ReactNode } from 'react'
import * as S from './styled'

type CardGridProps = {
  children: ReactNode
}

export default function CardGrid({ children }: CardGridProps): JSX.Element {
  return <S.Wrapper>{children}</S.Wrapper>
}
