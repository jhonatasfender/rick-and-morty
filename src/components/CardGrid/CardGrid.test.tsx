import { renderWithThemeAndRoute } from '../../utils/helpers'
import CardGrid from './'

describe('<CardGrid />', () => {
  it('should load the component with default values', () => {
    const { container } = renderWithThemeAndRoute(<CardGrid>test</CardGrid>)

    expect(container).toMatchSnapshot()
  })
})
