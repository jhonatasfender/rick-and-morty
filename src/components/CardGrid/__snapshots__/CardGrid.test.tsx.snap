// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardGrid /> should load the component with default values 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1rem;
  padding: 1rem;
}

<div>
  <div
    class="c0"
  >
    test
  </div>
</div>
`;
